pkg/api/pod/util.go

```diff
@@ -1275,19 +1275,35 @@ func MarkPodProposedForResize(oldPod, newPod *api.Pod) {
 		return
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) && len(newPod.Spec.InitContainers) != len(oldPod.Spec.InitContainers) {
+		return
+	}
+
 	for i, c := range newPod.Spec.Containers {
 		if c.Name != oldPod.Spec.Containers[i].Name {
 			return // Update is invalid (container mismatch): let validation handle it.
 		}
-		if c.Resources.Requests == nil {
-			continue
-		}
-		if cmp.Equal(oldPod.Spec.Containers[i].Resources, c.Resources) {
+		if c.Resources.Requests == nil || cmp.Equal(oldPod.Spec.Containers[i].Resources, c.Resources) {
```

Comment:
use `apiequality.Semantic.DeepEqual` instead of `cmp.Equal` (same below)

see https://github.com/kubernetes/kubernetes/issues/104821

Reply:
Done.